### ISSUE TITLE
Chats no longer refresh since recent update, requires "backing out" or force close to refresh

### ISSUE THREAD

**Issue by binarydad** [vector-im/element-android#7876]
Since around 1.5.12 I believe (at least 1.5.14), I've noticed that when I open the app, the chats no longer refresh, and I'm seeing old messages from the last time I used the app. I have to "back out", force close, or clear the cache (in Settings) for it to work again. Only then does the "rolling" green loading indicator cycle along the top of the chats and they begin refreshing. Still, even after this, when I open the app again later on, it has not refreshed since that time. I'm currently using 1.5.18 and the issue is still there. 

I am not seeing this on the desktop/web clients. 

I'm using Synapse 1.74 on Ubuntu. The device is Android 13 on a Galaxy S22. Thanks!

**Comment by bmarty** [comment]
Hi, thanks for the feedback @binarydad .
It looks like your app is not syncing when in background (and even when in foreground, which is more concerning).
At first check, can you navigate to `Settings/Notifications/Troubleshoot Notification` and see what's happening there?

**Comment by q234rty** [comment]
This happened to me as well, starting from 1.5.14 (1.5.13 worked fine).

**Comment by q234rty** [comment]
I believe the sync related changes caused the app to not try to sync at all in certain cases, but not sure.

**Comment by ryanpeters-MSFT** [comment]
My check of the notifications was fine. I did notice, however, that after clearing cache again, it seems to be "keeping up" with the state when comparing it to the desktop app. I'll monitor it and report back if it starts falling behind in terms of chat state/messages.

**Comment by binarydad** [comment]
Update since last time... it seems oddly intermittent. About 80% of the time, going back into the app, I immediately see the green bar sliding/loading across the top, and chats refresh. However, I just now went back into the app and it still had chat state from very early this morning (backing out of the app and re-opening caused them to refresh). 

Still, this was almost never a problem before some of the recent updates (not sure which one). 

How can I assist with this further? Thanks!

**Comment by q234rty** [comment]
In my experience this only happens if you leave the app in the background long enough and then go back to the app.

**Comment by binarydad** [comment]
Understood. Either way, it's a new (only slightly annoying) behavior.

**Comment by binarydad** [comment]
> Hi, thanks for the feedback @binarydad . It looks like your app is not syncing when in background (and even when in foreground, which is more concerning). At first check, can you navigate to `Settings/Notifications/Troubleshoot Notification` and see what's happening there?

FYI, still happening - I'll open the app and I only see messages from the day before, until I back out of the app. Is there anything else I can try or give additional information? Thanks.

**Comment by binarydad** [comment]
Would the PR (https://github.com/vector-im/element-android/pull/7724) possibly address this issue?

**Comment by binarydad** [comment]
Sorry for being noisy. Still having the same behavior.

**Comment by joshuaboniface** [comment]
Another confirmation of this behaviour., though slightly different. In my case, it also affects incoming push notifications. Those do not come in at all until I open the app and it refreshes. But the app does refresh itself once opened within a few seconds.

Troubleshooting notifications hangs during the "Test Push" stage. Confirming on my homeserver that nothing comes in beyond the earlier test that hits the pushers endpoint.

```
Jan 16 10:01:35 synapse matrix-synapse[3320551]: 2023-01-16 10:01:35,525 - synapse.access.http.8008 - 460 - INFO - GET-8 - ::ffff:x.x.x.x - 8008 - {@joshuaboniface:bonifacelabs.ca} Processed request: 0.023sec/0.001sec (0.005sec, 0.002sec) (0.002sec/0.016sec/3) 344B 200 "GET /_matrix/client/r0/pushers HTTP/1.1" "Element/1.5.20 (samsung SM-G970W; Android 12; SP1A.210812.016.G970WVLS8IVK2; Flavour GooglePlay; MatrixAndroidSdk2 1.5.20)" [0 dbevts]
```

If I have someone send a real push, my pusher worker just shows:

```
Jan 16 10:15:22 synapse0.p.bonilan.net matrix-synapse-pusher[3324368]: 2023-01-16 10:15:22,122 - synapse.push.httppusher - 225 - INFO - httppush.process-18 - Processing 1 unprocessed push actions for @joshuaboniface:bonifacelabs.ca/im.vector.app.android/xxx:xxx-xxx starting at stream_ordering 10084081
Jan 16 10:15:22 synapse0.p.bonilan.net matrix-synapse-pusher[3324368]: 2023-01-16 10:15:22,387 - synapse.http.client - 460 - INFO - httppush.process-18 - Received response to POST https://matrix.org/_matrix/push/v1/notify: 200           
```

I can't tell for certain if this is something wrong with the homeserver, or something wrong with the mobile client, but it's quite annoying in either case.

**Comment by binarydad** [comment]
> Another confirmation of this behaviour., though slightly different. In my case, it also affects incoming push notifications. Those do not come in at all until I open the app and it refreshes. But the app does refresh itself once opened within a few seconds.
> 
> Troubleshooting notifications hangs during the "Test Push" stage. Confirming on my homeserver that nothing comes in beyond the earlier test that hits the pushers endpoint.
> 
> ```
> Jan 16 10:01:35 synapse matrix-synapse[3320551]: 2023-01-16 10:01:35,525 - synapse.access.http.8008 - 460 - INFO - GET-8 - ::ffff:x.x.x.x - 8008 - {@joshuaboniface:bonifacelabs.ca} Processed request: 0.023sec/0.001sec (0.005sec, 0.002sec) (0.002sec/0.016sec/3) 344B 200 "GET /_matrix/client/r0/pushers HTTP/1.1" "Element/1.5.20 (samsung SM-G970W; Android 12; SP1A.210812.016.G970WVLS8IVK2; Flavour GooglePlay; MatrixAndroidSdk2 1.5.20)" [0 dbevts]
> ```
> 
> This was tested with push integrated into the main process as a test; I switched back to the dedicated pusher worker, and then if I have someone send a real push, my pusher worker just shows:
> 
> ```
> Jan 16 10:15:22 synapse matrix-synapse-pusher[3324368]: 2023-01-16 10:15:22,122 - synapse.push.httppusher - 225 - INFO - httppush.process-18 - Processing 1 unprocessed push actions for @joshuaboniface:bonifacelabs.ca/im.vector.app.android/xxx:xxx-xxx starting at stream_ordering 10084081
> Jan 16 10:15:22 synapse matrix-synapse-pusher[3324368]: 2023-01-16 10:15:22,387 - synapse.http.client - 460 - INFO - httppush.process-18 - Received response to POST https://matrix.org/_matrix/push/v1/notify: 200           
> ```
> 
> I can't tell for certain if this is something wrong with the homeserver, or something wrong with the mobile client, but it's quite annoying in either case.
> 
> Worth noting that this affects myself on Android, but does not seem to affect other users on iOS, nor users on the desktop clients.

Same here as well. After noticing it didn't refresh, I tested sending a message to myself from my son's phone to see if I was at least receiving notifications in some fashion, but they did not come (they were received fine on web/desktop). And as @joshuaboniface mentioned, after backing out and letting the app refresh, it all works again. It's something being idle after a while when it seems to emit the behavior.

**Comment by binarydad** [comment]
I'm running 1.5.22 and it is not yet fixed.


------ Original Message ------
From "opusforlife2" ***@***.***>
To "vector-im/element-android" ***@***.***>
Cc "Ryan Peters" ***@***.***>; "Mention" 
***@***.***>
Date 1/27/2023 4:14:24 PM
Subject Re: [vector-im/element-android] Chats no longer refresh since 
recent update, requires "backing out" or force close to refresh (Issue 
#7876)

**Comment by Tycho-S** [comment]
I'm still having this too. It's very annoying because even though notifications come in, the history shown in Element is stale. The sync thread just stays on idle. I have to force kill the app too make it work again but it will go back to idle after a while and then it remains dead.

I've found a workaround: using the F-droid version of element which doesn't use push. That keeps the sync thread alive somehow. But it absolutely kills my battery life because it constantly has to sync in the background. I see in this thread that others have discovered this workaround as well.

This is on Android 13 by the way. If you need any info please let met know.

**Comment by Tycho-S** [comment]
I'm still having this too. It's very annoying because even though notifications come in, the history shown in Element is stale. The sync thread just stays on idle. I have to force kill the app too make it work again but it will go back to idle after a while and then it remains dead. I was using the normal Google firebase cloud messaging by the way, not ntfy.. I have Google play installed but am not logged into it. This is not normally a problem for push.

I've found a workaround: using the F-droid version of element which doesn't use push. That keeps the sync thread alive somehow. But it absolutely kills my battery life because it constantly has to sync in the background. I see in this thread that others have discovered this workaround as well.

This is on Android 13 by the way. If you need any info please let met know.

**Comment by z411** [comment]
Same here, SyncThread changes to idle. Also I don't always get timely notifications or voice calls, although that might be because of MIUI (another friend with MIUI can confirm as well). It's not reliable for 1:1 communications as these sometimes are urgent. I sent some rageshakes with the "Not syncing" description.

**Comment by Tycho-S** [comment]
I've tried reverting to 1.5.13 and it seems to have resolved the issue. I think there is merit in what @q234rty suggested, that the hotfix in 1.5.14 ( https://github.com/vector-im/element-android/pull/7827 ) is causing this. It seems to be causing an edge case that stops the sync thread from continuing after a while.

**Comment by binarydad** [comment]
I've noticed this might be fixed in 1.5.24. Maybe with this commit? https://github.com/vector-im/element-android/pull/8019

**Comment by z411** [comment]
I just hit the bug again on 1.5.22 - I'll try 1.5.24 now.

**Comment by joshuaboniface** [comment]
Still having this issue on 1.5.24 as well. For fun I decided to also change the power management setting on the app (just in case it was Android backgrounding it) to Unrestricted,  and this makes no difference either.

**Comment by binarydad** [comment]
Interesting. I have yet to see it and I think I would have by now.

**Comment by Tycho-S** [comment]
I think it's because some phones kill the app in the background a lot. For me it often takes a day or two to manifest. If the app never runs that long it might not happen.

**Comment by Tycho-S** [comment]
@opusforlife2 I meant that this is perhaps why it might *not* manifest itself for some people like @binarydad . I'm also seeing the problem myself and I agree with you that it never happened before 1.5.14.

**Comment by cpot** [comment]
Same problem here 1.5.22, no E2E, homeserver
Terrible bug let's face it

**Comment by z411** [comment]
It hasn't happened to me since I updated to 1.5.24 but it might be coincidence.

**Comment by BloodyIron** [comment]
Yeah my SO is having very frequent problems every few days of just not receiving my messages, until they force stop and relaunch the app. As soon as they do, they get my messages. This really is a new bug, but it is unclear when it started.

**Comment by Tycho-S** [comment]
It's still broken for me too. Still happens a lot in fact. Perhaps it has to do with the fact that my instance isn't publicly reachable. I connect via VPN. I guess this has to do with why this bug manifests itself more often for me.

1.5.13 is working perfectly though. So I've stayed on that. I'm pretty sure the hotfix https://github.com/vector-im/element-android/pull/7827 caused an edge case because it's the only thing that changed in 1.5.14. And it was explicitly to do with the sync thread.

I've stopped testing new versions regularly now.

**Comment by BloodyIron** [comment]
This issue persists for my SO. No VPN involved.

**Comment by timokoesters** [comment]
At least on my device, it's fixed on Element 1.5.26. It was broken on 1.5.22. I haven't checked which particular version fixed it.

**Comment by Sir-Photch** [comment]
Still happening to me. I get:

```
Sync thread: Running(afterPause=true)
Sync request: Parsing 11 room(s) 0 toDevice(s)
```

And force-closing the app doesn't fix this.
I am on 1.5.32 with ntfy using dendrite on 0.12.0.

Interestingly, messages I send are received, but I don't get the replies.

**Comment by maxamillion** [comment]
Still happening here on 1.6.5 on Galaxy S23. I pay for Element One and it's basically a waste of money now but I'm holding out hope this gets fixed because I want the future of Matrix to come true.

**Comment by maxamillion** [comment]
Running FluffyChat on Android for now, it's actually really nice too.

**Comment by Tycho-S** [comment]
Still happening for me too. I am back on 1.5.13 which is the last version that actually worked.

**Comment by MattyBoombalatty** [comment]
I'm also having this issue. Android 13, Element version 1.6.5. Desktop client works fine. Might revert to an older version until this is fixed. It's pretty terrible.

**Comment by Tycho-S** [comment]
I don't think it'll ever be fixed. It was introduced by this hotfix: https://github.com/vector-im/element-android/pull/7827 and I tried to link to that but it's bot getting any attention.

But the good thing is that Element X is coming and it'll be a complete rewrite. It's not full-functional enough for me to use yet but hopefully it will be soon.

**Comment by MattyBoombalatty** [comment]
I don't think Element X will be up to speed for awhile. I'll check that version instead. Thank you.

**Comment by MattyBoombalatty** [comment]
I'm not able to find the v1.5.13 APK. Is it somewhere in the releases?

**Comment by Tycho-S** [comment]
Yes here: https://github.com/vector-im/element-android/releases/tag/v1.5.13

**Comment by MattyBoombalatty** [comment]
Thank you. I'm on 1.5.32 from F-Droid right now and it seems to be working alright thus far but I'll download this one just for the future. I appreciate your help.

**Comment by MattyBoombalatty** [comment]
Had to revert to v1. 5.13 as 1.6.6 still has this problem. Oh well. Hopefully the Element X team will be able to crank out features for the new interface.

**Comment by animegrafmays** [comment]
they wont, element x is cringe and barely usable for basic  conversation

**Comment by animegrafmays** [comment]
b-buh muh features
there are none. sliding sync is a waste and element x is not going anywhere. you literally cant use it a year later

**Comment by BloodyIron** [comment]
@animegrafmays please stay on topic. And we really don't need 3x comments that don't actually contribute to the original topic. This isn't an effective (or appropriate) place for comments of that nature. Best to post them in a place that will actually be relevant for them ;P (namely, not this thread)

**Comment by austinbutler** [comment]
Still happens on the current Play Store version, 1.6.14. My usage is about as vanilla as one could expect, Element, Matrix.org homeserver, Google Play services on a stock Pixel phone. Push notifications seem pretty reliable. Actually accessing the messages that clearly exist on the server is another story entirely. I'll click the push notification then be taken to the conversation, then I wait. And wait. And wait. The loading bar at the top scrolls forever but nothing happens. Sometimes backing out of the conversation helped. Sometimes I had to force stop.

Eventually nothing worked so I tried Element X again. At first it seemed like they'd fixed it there, but then it happened to me today on X, I had to force stop to get new messages I saw via push notification. My current setup is having both Element Android and Element X installed with Element Android notifications turned off and X notifications turned on. I'm now considering installing a third client like FluffyChat to see if anything can reliably do something as simple as retrieve the latest messages 🤕.
